src/pages/home/report/ReportActionsView.js

```diff
@@ -256,6 +257,15 @@ class ReportActionsView extends React.Component {
         return Visibility.isVisible() && !isSidebarCoveringReportView;
     }
 
+    // If the report is optimistic (AKA not yet created) we don't need to call openReport again
+    openReportIfNecessary() {
+        if (this.props.report.isOptimisticReport) {
+            return;
+        }
+
+        Report.openReport(this.props.report.reportID);
+    }
+
```

Comment:
I still think we should move this conditional to [openReport](https://github.com/Expensify/App/blob/1f3362988a06ef226cc0cf5e67d251d886159068/src/libs/actions/Report.js#L698)

Reply:
The existing comment got dismissed but I will repeat it here. I think that it's confusing to have `openReport` sometimes not open the report, but specifically in this case:
https://github.com/Expensify/App/pull/11439/files#diff-f7eec01ae30248000c28c7d3f68c7ea00ba10e5fc64f2854474a350beb2e9556R147

We are passing the (newly created) *optimistic* report to `openReport` there, and we want it to open in that case. Adding an `isOptimistic` flag doesn't really work because sometimes we actually want to open the optimistic report.

What we're talking about is passing an `actuallyOpenTheReport` flag to `OpenReport` which seems obtuse.

We could also just do something like:
```
if (this.props.report.isOptimisticReport) {
            Report.openReport(this.props.report.reportID);
}
```

Instead of creating this function, but I think having this function in one place is clearer because we can have a single comment explaining it.

But basically, simply knowing whether the report is optimistic or not is not enough information to know if we neeed to open it.

Reply:
Ah I see, because sometimes we want to create a new optimistic report, and sometimes we have are looking at an optimistic report and therefore we don't need to open it. `openReportIfNecessary` makes sense.

Reply:
Thanks for the additional context. I think that makes sense in this case then.